src/react-components/preferences-screen.js

```diff
@@ -1247,6 +1224,102 @@ class PreferencesScreen extends Component {
             prefType: PREFERENCE_LIST_ITEM_TYPE.CHECK_BOX
           }
         ]
+      ],
+      [
+        CATEGORY_GRAPHICS,
+        [
+          { key: "maxResolution", prefType: PREFERENCE_LIST_ITEM_TYPE.MAX_RESOLUTION },
+          {
+            key: "materialQualitySetting",
+            prefType: PREFERENCE_LIST_ITEM_TYPE.SELECT,
+            options: [
+              {
+                value: "low",
+                text: intl.formatMessage({
+                  id: "preferences-screen.material-quality-setting.low",
+                  defaultMessage: "Low"
+                })
+              },
+              {
+                value: "medium",
+                text: intl.formatMessage({
+                  id: "preferences-screen.material-quality-setting.medium",
+                  defaultMessage: "Medium"
+                })
+              },
+              {
+                value: "high",
+                text: intl.formatMessage({
+                  id: "preferences-screen.material-quality-setting.high",
+                  defaultMessage: "High"
+                })
+              }
+            ],
+            promptForRefresh: true
+          },
+          {
+            key: "enableDynamicShadows",
+            prefType: PREFERENCE_LIST_ITEM_TYPE.CHECK_BOX
+          },
+          {
+            key: "disableAutoPixelRatio",
+            prefType: PREFERENCE_LIST_ITEM_TYPE.CHECK_BOX
+          },
+          {
+            key: "enablePostEffects",
+            prefType: PREFERENCE_LIST_ITEM_TYPE.CHECK_BOX,
+            promptForRefresh: true
+          },
+          {
+            key: "enableBloom",
+            prefType: PREFERENCE_LIST_ITEM_TYPE.CHECK_BOX,
+            disableIfFalse: "enablePostEffects",
+            promptForRefresh: true
+          },
+          {
+            key: "aaMode",
+            prefType: PREFERENCE_LIST_ITEM_TYPE.SELECT,
+            disableIfFalse: "enablePostEffects",
+            promptForRefresh: true,
```

Comment:
Do `enableBloom` and `aaMode` toggles really require the refresh? Can't we skip the effect passes if they are turned off?

Reply:
I think both of these could probably be made to work but it seemed simpler just to have users refresh for now. Toggling the main option definitely requires a refresh since it changes how we initialize our WebGL context.